src/doc/grammar.md

```diff
@@ -512,7 +512,7 @@ unit_expr : "()" ;
 ```antlr
 struct_expr : expr_path '{' ident ':' expr
                       [ ',' ident ':' expr ] *
-                      [ ".." expr ] '}' |
+                      [ ".." expr ] ? '}' |
```

Comment:
But yeah I'll fix the rule to be valid, but this will be affected by my other PR.

I think I'll just close this down and do some more thought out work on the grammar.